pkg/clusteragent/admission/mutate/agent_sidecar/agent_sidecar.go

```diff
@@ -155,6 +171,14 @@ func (w *Webhook) injectAgentSidecar(pod *corev1.Pod, _ string, _ dynamic.Interf
 	if !agentSidecarExists {
 		agentSidecarContainer := w.getDefaultSidecarTemplate()
 		pod.Spec.Containers = append(pod.Spec.Containers, *agentSidecarContainer)
+		if w.isReadOnlyRootFilesystem() {
+			w.addDefaultSidecarSecurity(agentSidecarContainer)
+			pod.Spec.Volumes = append(pod.Spec.Volumes, *w.getDefaultSidecarVolumeTemplate())
+			// Don't want to apply any overrides to the agent sidecar init container
+			defer func() {
+				pod.Spec.InitContainers = append(pod.Spec.InitContainers, *w.getDefaultSidecarInitTemplate())
```

Comment:
Suggestion: 

pass the volume name as an argument to `getDefaultSidecarInitTemplate()`

You can then do something like this:

```
configVolume := w.getDefaultSidecarVolumeTemplate()
pod.Spec.Volumes = append(pod.Spec.Volumes, *configVolume)
configInitContainer := w.getDefaultSidecarInitTemplate(configVolume.Name)
pod.Spec.InitContainers = append(pod.Spec.InitContainers, configInitContainer)
```

This can help manually writing the volume name in different places, and makes the function easier to test.